### ISSUE TITLE
Typescript definitions

### ISSUE THREAD

**Issue by radicand** [keajs/kea#65]
Are there any plans to create TypeScript definitions for kea, or are there any recommended filler definitions to use until then?

**Comment by mariusandra** [comment]
Hey, I don't use TS myself, so I can't comment on this. I'm happy to accept community contributions on this matter though.

**Comment by mariusandra** [comment]
TypeScript support is coming to Kea!

Unfortunately [as predicted by this commenter on reddit](https://www.reddit.com/r/reactjs/comments/d386wp/kea_10_released_data_layer_for_react_powered_by/f00ddmv/), it's impossible to automatically get full type support on Kea. I tried a lot of different options (and will write a blog post about them later) before settling on [`kea-typegen`](https://github.com/keajs/kea-typegen).

Running `kea-typegen` will use the TypeScript Compiler API to parse your `kea()` calls and automatically generate `logicType.ts` files for them. 

This is all still very raw, but it's already really useful provided you're using the [supported subset](https://github.com/keajs/kea-typegen/projects/1) of the API. A lot of things are already supported and I'm adding more every day.

[Here's a screencast](https://www.youtube.com/watch?v=jGy-p9UxcBA) I made that showcases the support as it stands now. I apologise for the bad audio: It's the first screencast I have ever made.

Any and all feedback is absolutely welcome!
